### PR TITLE
[DM | Deserialization] Support for index variables

### PR DESCRIPTION
**Pull request by NicL9923**

### CODE REVIEW

File: libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts [hunk 1]
```diff
@@ -312,7 +323,43 @@ const createConnections = (
   }
 
   if (isLoop && sourceNode && destinationNode) {
-    addParentConnectionForRepeatingElements(destinationNode, sourceNode, sourceSchemaFlattened, targetSchemaFlattened, connections);
+    let indexFnRfKey: string | undefined = undefined;
+
+    // TODO: Ensure support for nested loops w/ index variables
```
Comment (refortie) [inline libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts:328]: Still TBD?

File: libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts [hunk 2]
```diff
@@ -312,7 +323,43 @@ const createConnections = (
...
+
+    // Loop index variable handling (create index() node, match up variables to respective nodes, etc)
+    const idxOfIdxVariable = targetKey.replaceAll('$for', '').lastIndexOf('$');
```
Comment (refortie) [inline libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts:331]: $for -> mapNodeParams.for

File: libs/data-mapper/src/lib/utils/DataMap.Utils.ts [hunk 3]
```diff
@@ -175,12 +175,14 @@ export const splitKeyIntoChildren = (sourceKey: string): string[] => {
   return results;
 };
 
+export const getSourceLoopKey = (targetKey: string): string => {
+  const forArgs = targetKey.substring(targetKey.lastIndexOf('$for(') + 5, targetKey.lastIndexOf(')'));
```
Comment (refortie) [inline libs/data-mapper/src/lib/utils/DataMap.Utils.ts:179]: $for -> mapNodeParams.for

`targetKey.lastIndexOf(mapNodeParams.for) + mapNodeParams.for.length + 1`?

**Review by refortie — APPROVED** [review]

File: libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts (hunk 1 above)
Comment (NicL9923) [reply libs/data-mapper/src/lib/mapDefinitions/MapDefinitionDeserializer.ts:328]: Yeup, taking a look today

**Pull request merged**